tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/contact/EmailAddressContact.scala

```diff
@@ -36,16 +35,19 @@ case class ContactFields(address: MailAddress, firstname: String = "", surname:
     address.asString().contains(part) || firstname.contains(part) || surname.contains(part)
 }
 
-case class AccountEmailContact(accountId: String, id: UUID, address: MailAddress) {
-  def this(accountId: AccountId, emailAddressContact: EmailAddressContact) =
-    this(accountId.getIdentifier, emailAddressContact.id, emailAddressContact.fields.address)
-}
-
 trait EmailAddressContactSearchEngine {
   def index(accountId: AccountId, fields: ContactFields): Publisher[EmailAddressContact]
 
   def index(domain: Domain, fields: ContactFields): Publisher[EmailAddressContact]
 
+  def update(accountId: AccountId, mailAddress: MailAddress, updatedFields: ContactFields): Publisher[EmailAddressContact]
+
+  def update(domain: Domain, mailAddress: MailAddress, updatedFields: ContactFields): Publisher[EmailAddressContact]
```

Comment:
I think when talking about it with @chibenwa , we came to the conclusion that for example for well lsc sync with ldap if there is an email change, it would be translated by a delete and a create (your case). 

But then after I wasn't sure about what the third party app would send... If an update of mailaddress only was possible or not. So I did it in case.

But if we are sure that we only update the name, and in case the email change we are ok to delete and recreate the contact, then I'm ok to simplify this method, will make the logic easier

Reply:
If we are not sure how the update works with third-party apps, maybe using Option for mailAddress is better for now?

>  if we are sure that we only update the name, and in case the email change we are ok to delete and recreate the contact, then I'm ok to simplify this method, will make the logic easier

Otherwise, I am OK with this.

Reply:
Option would make sense in this case yes... not gonna simplify the code though :D

Reply:
I tried something, let me know if it's ok for you!

Reply:
Ok been agreed in grooming to simplify this and only reduce the update to name update.

If the mail address changes -> delete (old email address) + create (new email address)